test/extended/openstack/loadbalancers.go

```diff
@@ -54,101 +56,103 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb] The Openstack
-		if lbProvider != "amphora" {
-			e2eskipper.Skipf("Test not applicable for LoadBalancer provider different than Amphora: %q", lbProvider)
-		}
-
-		g.By("Creating Openshift resources")
-		// Reference on test/extended/testdata/openstack/udplb-default-manifests.yaml
-		labels := map[string]string{"app": "udp-lb-default-dep"}
-		depName := "udp-lb-default-dep"
-		svcName := "udp-lb-default-svc"
-		testDeployment := e2edeployment.NewDeployment(depName, 2, labels, "udp-test",
-			imageutils.GetE2EImage(imageutils.Agnhost), appsv1.RollingUpdateDeploymentStrategyType)
-		testDeployment.Spec.Template.Spec.SecurityContext = e2epod.GetRestrictedPodSecurityContext()
-		testDeployment.Spec.Template.Spec.Containers[0].SecurityContext = e2epod.GetRestrictedContainerSecurityContext()
-		testDeployment.Spec.Template.Spec.Containers[0].Args = []string{"netexec", "--udp-port=8081"}
-		testDeployment.Spec.Template.Spec.Containers[0].Ports = []v1.ContainerPort{{
-			ContainerPort: 8081,
-			Protocol:      "UDP",
-		}}
-		deployment, err := clientSet.AppsV1().Deployments(oc.Namespace()).Create(context.TODO(),
-			testDeployment, metav1.CreateOptions{})
-		o.Expect(err).NotTo(o.HaveOccurred())
-		err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName)
-		jig.Labels = labels
-		svc, err := jig.CreateLoadBalancerService(5*time.Minute, func(svc *v1.Service) {
-			svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: 8082, TargetPort: intstr.FromInt(8081)}}
-			svc.Spec.Selector = labels
-		})
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By("Checks from openshift perspective")
-		o.Expect(svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]).ShouldNot(o.BeEmpty(),
-			"load-balancer-id annotation missing")
-		loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
-		o.Expect(svc.Status.LoadBalancer.Ingress[0].IP).ShouldNot(o.BeEmpty(), "FIP missing on svc Status")
-		o.Expect(svc.Spec.ExternalTrafficPolicy).Should(o.Equal(v1.ServiceExternalTrafficPolicyTypeCluster),
-			"Unexpected ExternalTrafficPolicy on svc specs")
-
-		g.By("Checks from openstack perspective")
-		lb, err := octavialoadbalancers.Get(loadBalancerClient, loadBalancerId).Extract()
-		o.Expect(err).NotTo(o.HaveOccurred())
-		o.Expect(lb.Provider).Should(o.Equal("amphora"), "Unexpected provider in the Openstack LoadBalancer")
-
-		svcIp := svc.Status.LoadBalancer.Ingress[0].IP
-		svcPort := fmt.Sprint(svc.Spec.Ports[0].Port)
-		fip, err := getFipbyFixedIP(networkClient, lb.VipAddress)
-		o.Expect(err).NotTo(o.HaveOccurred())
-		o.Expect(fip.FloatingIP).Should(o.Equal(svcIp), "Unexpected floatingIp in the Openstack LoadBalancer")
-		o.Expect(lb.Pools).Should(o.HaveLen(1), "Unexpected number of pools on Openstack LoadBalancer %q", lb.Name)
+	for _, i := range availableLbProvidersUnderTests {
```

Comment:
I don't know why but your proposal did not work. I tried that, but if I don't set the variable later, the loop is not working as expected. I found this approach on origin repo, for example: https://github.com/openshift/origin/blob/master/test/extended/apiserver/apply.go#L57

Reply:
Instead of looping like this shouldn't we just check the LB provider that's configured and then when LB is created in Octavia checking if it has correct provider?

Reply:
They're testing different providers really. But I can't argue with the concept of a separate test if you need it like this for your tools. The loop and skip just feels a bit weird here.

Do you need both tests to appear in every result set, even if skipped? If so, I guess `for` loop is the way to go. Otherwise you could check LB provider before `g.It(fmt.Sprintf("should create an UDP %s LoadBalancer when an UDP svc with type:LoadBalancer is created on Openshift", lbProviderUnderTest)`.

Reply:
Thanks! Done